pkg/console/controllers/route/controller.go

```diff
@@ -159,6 +186,48 @@ func (c *RouteSyncController) SyncRoute(operatorConfig *operatorsv1.Console) (co
 	return rt, rtIsNew, "", rtErr
 }
 
+// Custom route sync needs to:
+// 1. validate if the reference for secret with serving-cert is defined in operator config(in case a non-openshift cluster domain is used)
+// 2. sync the secret with the serving-cert, from the openshift-config,into the openshift-console namespace
+// 3. create the custom console route
+// 4. check custom route health
+// 5. validate the route
+func (c *RouteSyncController) SyncCustomRoute(operatorConfig *operatorsv1.Console) (consoleRoute *routev1.Route, isNew bool, reason string, err error) {
+	if configErr := routesub.ValidateCustomRoute(operatorConfig, c.routeClient); configErr != nil {
+		return nil, true, "InvalidCustomRouteConfig", configErr
+	}
+
+	if scscErr := c.SyncCustomServigCertSecret(operatorConfig); scscErr != nil {
+		return nil, false, "FailedSyncServingCertSecret", scscErr
+	}
+
+	cr, crIsNew, crErr := routesub.GetOrCreate(c.routeClient, routesub.CustomRoute(operatorConfig))
+	if crErr != nil {
+		return nil, false, "FailedCreateCustomRoute", crErr
+	}
+	c.CheckRouteHealth(operatorConfig, cr)
+
+	if _, changed := routesub.Validate(cr); changed {
+		return nil, true, "InvalidCustomRoute", customerrors.NewSyncError("custom route is invalid, correcting route state")
```

Comment:
So this deviates a little bit from the `Validate()` in `SyncRoute()`.  Looking into this a bit to make sure its ok.

Reply:
thought that since the `Validate()` is just checking the `To`, `Port`, `TLS`, `WildcardPolicy` then it cant hurt to check those on the custom route, since these should be the same